core/chains/evm/txmgr/builder.go

```diff
@@ -51,12 +51,13 @@ func NewTxm(
 	feeCfg := NewEvmTxmFeeConfig(fCfg)     // wrap Evm specific config
 	txmClient := NewEvmTxmClient(client)   // wrap Evm specific client
 	ethBroadcaster := NewEvmBroadcaster(txStore, txmClient, txmCfg, feeCfg, txConfig, listenerConfig, keyStore, txAttemptBuilder, txNonceSyncer, lggr, checker, chainConfig.NonceAutoSync())
-	ethConfirmer := NewEvmConfirmer(txStore, txmClient, txmCfg, feeCfg, txConfig, dbConfig, keyStore, txAttemptBuilder, lggr)
+	ethTracker := NewEvmTracker(txStore, lggr)
+	ethConfirmer := NewEvmConfirmer(txStore, txmClient, ethTracker, txmCfg, feeCfg, txConfig, dbConfig, keyStore, txAttemptBuilder, lggr)
 	var ethResender *Resender
 	if txConfig.ResendAfterThreshold() > 0 {
-		ethResender = NewEvmResender(lggr, txStore, txmClient, keyStore, txmgr.DefaultResenderPollInterval, chainConfig, txConfig)
+		ethResender = NewEvmResender(lggr, txStore, txmClient, keyStore, txmgr.DefaultResenderPollInterval, chainConfig, txConfig, ethTracker)
```

Comment:
nit: I noticed that for the evm resender here, we are sticking the tracker at the end of the arg list, but for the generic newResender on line 95 below, we are sticking it in the middle of the arg list. Is this just arbitrary or is there a reason to be putting it in different places in the arg list for the different constructors?

Taking a step back though, I don't love when you get these constructors with a large number of args and somewhat random ordering. I wonder if we could collect some of this arguments together into meaningful structs and use builders to put these things together and submit them to the various constructors that need them. That way we could be passing around fewer arguments and make the code a little more compact and readable.

Reply:
Nice catch, I'll move it to the correct position for consistency. I agree that it could be an improvement to pass a config struct around instead, but that's out of scope for this PR.